cmd/setPropertiesProcessor.go

```diff
@@ -22,6 +22,25 @@ package cmd
 
 import "github.com/Azure/azure-storage-azcopy/v10/common"
 
+// setting SetPropertiesAPIOption for choosing which API to use
+// TODO name change needed
+func setSetPropertiesAPIOption(cca *CookedCopyCmdArgs) common.SetPropertiesAPIOption {
+	if cca.metadata != "" {
+		if cca.blockBlobTier != common.EBlockBlobTier.None() {
+			return common.ESetPropertiesAPIOption.SetTierAndMetaData()
```

Comment:
What if there are three or more properties. The best way to do it using bit mask. Look for ways to do it.